todo_app/app.py

```diff
@@ -15,3 +15,13 @@ def add_todo_item():
     new_item_title = request.form.get('title')
     add_item(new_item_title)
     return redirect(url_for('index'))
+
+@app.route('/complete_item<id>')
+def mark_item_as_complete(id):
```

Comment:
I can see these routes but I can't see how they're triggered from your app - is the user expected to manually hit the API for these? It would be good to tie that to the front end (more details below)

I'd also instinctively add a `/` between the verb and the id:
```suggestion
@app.route('/complete_item/<id>')
def mark_item_as_complete(id):
```

(As a bonus, I think this makes more sense as a POST request since we're changing data - sadly we do have to choose between either GET or POST for HTML forms!)